### ISSUE TITLE
Update lang pack for 4.3.1

### ISSUE THREAD

**Issue by WatchTh1** [TechInvestLab/Archi-rus-pack#2]
Доброго! Не так давно вышла версия инструмента 4.3.1. Хотелось бы обновлённый русификатор. Готов сделать в принципе и сам)

С уважением, Лучков Александр.

**Comment by ailev** [comment]
Конечно, нужно делать! Помним, что самый последний русификатор был сделан тут -- https://github.com/smeagol74/archi-ru

**Comment by WatchTh1** [comment]
Я потихоньку сижу, перевожу. Хотелось бы понять как можно организовать
проверку перевода. Лучше пораньше.
Плюс есть вопросы с переводом ресурсов плагинов, но это лучше отдельно
обсудить.

ср, 16 янв. 2019 г., 0:15 ailev notifications@github.com:

**Comment by ailev** [comment]
Нужно спросить у @smeagol74 -- вдруг он тоже "сидит, переводит"? Опять же: можно переводить "с нуля", а можно просто доперевести то, что не было переведено ранее.

Про плагины я вообще не думал пока.

Проверка простая: запускается редактор и смотрится, что получилось. Ибо вне контекста каждая надпись мало информативна, в исходных кодах непонятно, что могло бы означать то или иное слово (я, когда переводил, постоянно проверял -- в каком контексте слово, запускал с русификацией).

**Comment by WatchTh1** [comment]
Я как раз доперевожу, с некоторыми изменениями иногда.

Перевод с нуля слишком много времени займёт.

Про плагины я вообще думаю написать реквест авторам Арчи, для организации
многоязычной поддержки, и вынос строк перевода в отдельные ресурсы, вместо
встраивания в бинарный код приложенич. Но это уже отдельная история.

ср, 16 янв. 2019 г., 13:14 ailev notifications@github.com:

**Comment by smeagol74** [comment]
> Нужно спросить у @smeagol74 -- вдруг он тоже "сидит, переводит"? Опять же: можно переводить "с нуля", а можно просто доперевести то, что не было переведено ранее.

Нет, я пока не смотрел чего там поменялось. Но переводить всё каждый раз "с нуля" совершенно необязательно. Достаточно поглядеть чего изменилось между версиями и внести соответствующие изменения в русские файлы с ресурсами. Я для этого как раз веду [проектик чистых "английских" ресурсов с тэгами по релизам Archi](https://github.com/smeagol74/archi-nls)

Тогда можно средствами `git` поглядеть разницу между тэгами командой `git diff 4.3.1 4.0.2` (это будет как раз разница с последней версии у меня до нынешнего релиза), увидеть чего изменилось в английской версии и внести соответствующие изменения в русские версии ресурсов. Про это у меня в инструкции к переводу написано.

@WatchTh1 буду крайне признателен, если Вы организуете pull request с дополнениями перевода до версии 4.3.1. Заранее спасибо.

**Comment by WatchTh1** [comment]
Собственно дифф-мержем и работаю) Пулл-реквест сделаю как будет готово. Как
именно обсудим отдельно.

Качество перевода я бы хотел, чтобы проверили люди отличные от меня.

И если получится, то сделать это итеративно, например пушить куда-то в
ветку, версию перевода сборкой плагина и/или исходниками nls-модуля.

ср, 16 янв. 2019 г., 18:11 Konstantin Borisov notifications@github.com:

**Comment by smeagol74** [comment]
> И если получится, то сделать это итеративно, например пушить куда-то в ветку,

На мой вкус, самый простой способ — это сделать fork от моей версии, внести изменения, сделать push и выдать ссылку на свой fork. Потом, как будете готовы, просто делаете pull-request своей версии форка.

Я могу глянуть без проблем. Я так понимаю, что Вы добрались до перевода пакета `com.archimatetool.help.ru/nl/ru` со всякими примерчиками и подсказочками? Только там вроде оставалось чего-то непереведённое, а между версиями я в основном только технические изменения вижу.

**Comment by WatchTh1** [comment]
Да, добрался, до него.
Форкнусь и дам ссылку на форк, проверимся там и сюда будем тащить уже
готовые бинарники с включенным переводом.

Согласны?

ср, 16 янв. 2019 г., 18:47 Konstantin Borisov notifications@github.com:

**Comment by WatchTh1** [comment]
Предварительная версия перевода для арчи, собранная и упакованная. Прошу по возможности посмотреть. 
Ожидаемые недостатки: 
 - Могут быть не переведены некоторые базовые компоненты и действия интерфейса Eclipse (например Undo/Redo);
 - Нет перевода хэлпа Archi;
 - Не встроен перевод подсказок.


Ссылка на архив с переводом:

[Archi-rus-pack-4.3.1.zip](https://github.com/TechInvestLab/Archi-rus-pack/files/2830155/Archi-rus-pack-4.3.1.zip)

Упаковано аналогично расположенному в этом репозитории.

**Comment by WatchTh1** [comment]
Выкладываю версию перевода для 4.3.1. Сохранены все подскази, встроен перевод базовых компонентов Eclipse.
Пока не сделал полный перевод справки, но его там оказывается и не было. Потихоньку продвигаю до версии 4.3.3.

[Archi-rus-pack-4.3.1.zip](https://github.com/TechInvestLab/Archi-rus-pack/files/2923369/Archi-rus-pack-4.3.1.zip)
